strategy-cache-only/service-worker.js

```diff
@@ -0,0 +1,34 @@
+var CACHE = 'cache-only';
+
+// On install, cache some resource.
+self.addEventListener('install', function(evt) {
+  console.log('The service worker is being installed.');
+  evt.waitUntil(precache());
+});
+
+// On fetch, use cache only strategy.
+self.addEventListener('fetch', function(evt) {
+  console.log('The service worker is serving the asset.');
+  evt.respondWith(fromCache(evt.request));
+});
+
+// Open a cache and use `addAll()` with an array of assets to add all of them
+// to the cache. Extend the installation until all the assets are saved.
+function precache() {
+  return caches.open(CACHE).then(function (cache) {
+    return cache.addAll([
+      './controlled.html',
+      './asset'
+    ]);
+  });
+}
+
+// As simple as opening the proper cache and search for the requested resource.
```

Comment:
"simple" is very relative. I'd suggest rewriting this comment as something like 'We will open the cache where we stored the data before, and search for the requested resource'